app/controllers/api/v1/punches_controller.rb

```diff
@@ -0,0 +1,16 @@
+module Api
+  module V1
+    class PunchesController < ApiController
+      def create
+        created_punches = CreatePunchesService.call(create_punches_params, current_user)
+        render json: created_punches, status: :created
+      end
+
+      private
+
+      def create_punches_params
+        params.permit(punches: [:from, :to, :project_id]).require(:punches)
```

Comment:
you can do params.require(:punches).permit([:from, :to, :project_id])

Reply:
sorry I showed the wrong syntax it should be `params.require(:punches).permit(:from, :to, :project_id)`

Reply:
https://github.com/rails/strong_parameters#strong-parameters

Reply:
The approach is correct, in fact, he's using `require` only to return the value into the "punches" attributes. :stuck_out_tongue: